Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpReceiverBackend.cpp

```diff
@@ -53,10 +53,10 @@ Vector<RTCRtpSynchronizationSource> GStreamerRtpReceiverBackend::getSynchronizat
 Ref<RealtimeMediaSource> GStreamerRtpReceiverBackend::createSource(const String& trackKind, const String& trackId)
 {
     if (trackKind == "video"_s)
-        return RealtimeIncomingVideoSourceGStreamer::create(trackId.isolatedCopy());
+        return RealtimeIncomingVideoSourceGStreamer::create(AtomString { trackId.isolatedCopy() });
```

Comment:
This idiom seems peculiar. Typically the point of calling `isolatedCopy` is to make a new string that can be passed to another thread without any race conditions involving its reference count. But converting to an atom string immediately makes it unsafe to pass to a new thread once again. Do you understand the benefit of the isolatedCopy here?

Reply:
I asked @philn why he did that but he doesn't remember. Personally, I suspect it isn't needed but I tried to be conservative.

Reply:
@philn suspect he did that as a way to get an r-value reference because that's what the function expects. It is an expensive way to get an r-value reference but that sounds plausible and again hints that this is likely not needed.

Reply:
Yes, if that was the reason then it was definitely a mistake. I always worry about making our core classes too hard to use correctly and seems Iike this is small example of that. We renamed this from `copy` to `isolatedCopy` and it was to prevent people from using it in cases like that unnecessarily. Our goal once strings were immutable was to eliminate unnecessary copying.

Reply:
Yeah, sorry about this. Thanks @cdumez for fixing this ❤️

Reply:
I feel like crossThreadCopy() would be clearer than isolatedCopy() personally.

Note that we already have CrossThreadCopier and a crossThreadCopy() free function.